### DISCUSSION TITLE
74HC595 support for RP2040

### DISCUSSION THREAD

**Discussion by Hyakutake12** [Paciente8159/uCNC#543] [Ideas]
Hi!
I see shift register outputs are not supported for rp2040 mcu. Any plans to implement this feature?

**Comment by Paciente8159** [comment]
Shift register is supported for all mcu but is bit banged.
I can add support for I2S for RP2040.
I will add this as a feature request.
Is it for a custom board or an existing board?

**Reply by Hyakutake12** [reply]
It is for a custom board using pi pico. Will be grateful to little guidance configuring 74hc595 in rp2040, since i couldn't get the right override files generated from the web tool.. thanks

**Reply by Paciente8159** [reply]
I've removed the limitation on the web builder that prevent configuring the 74hc595 offsets for the pins in all MCU (was locked to ESP32 only), so you should be able to do a usable configuration now.

I also remembered it's possible to use SPI for this task if you don't have intention of using SPI for any other application.
